src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -1272,8 +1273,8 @@ bool PhaseIdealLoop::loop_predication_impl_helper(IdealLoopTree* loop, IfProjNod
     // upper bound test. We always need to create skeleton predicates in order to properly remove dead loops when later
     // splitting the predicated loop into (unreachable) sub-loops (i.e. done by unrolling, peeling, pre/main/post etc.).
     IfTrueNode* template_assertion_predicate_proj =
-        add_template_assertion_predicate(iff, loop, hoisted_check_proj, parse_predicate_proj, upper_bound_proj, scale,
-                                         offset, init, limit, stride, rng, overflow, reason);
+        create_template_assertion_predicate(if_opcode, cl, parse_predicate_proj, upper_bound_proj, scale, offset, range,
+                                            deopt_reason);
```

Comment:
We only use the opcode from the `iff`. `init`, `limit` and `stride` can be fetched from the `CountedLoop` again.